MantidQt/CustomInterfaces/src/Reflectometry/ReflMainWindowPresenter.cpp

```diff
@@ -0,0 +1,130 @@
+#include "MantidQtCustomInterfaces/Reflectometry/ReflMainWindowPresenter.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflMainWindowView.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflRunsTabPresenter.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflSettingsTabPresenter.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+/** Constructor
+* @param view :: [input] The view we are managing
+* @param runsPresenter :: [input] A pointer to the 'Runs' tab presenter
+* @param settingsPresenter :: [input] A pointer to the 'Settings' tab presenter
+*/
+ReflMainWindowPresenter::ReflMainWindowPresenter(
+    IReflMainWindowView *view, IReflRunsTabPresenter *runsPresenter,
+    IReflSettingsTabPresenter *settingsPresenter)
+    : m_view(view), m_runsPresenter(runsPresenter),
+      m_settingsPresenter(settingsPresenter) {
+
```

Comment:
I'm not sure about this. In the `initLayout()` method, I have:

```
void QtReflMainWindowView::initLayout() {
  m_ui.setupUi(this);

  // Create the tabs
  auto runsPresenter = createRunsTab();
  auto settingsPresenter = createSettingsTab();

  // Create the presenter
  m_presenter.reset(
      new ReflMainWindowPresenter(this, runsPresenter, settingsPresenter));
}
```

If I don't pass `runsPresenter` and `settingsPresenter` as arguments to the `ReflMainWindowPresenter`'s constructor, the `QtReflMainWindowView` would need to keep pointers to both `IReflRunsTabPresenter` and `IReflSettingsTabPresenter`, so that I can later do:

```
IReflRunsTabPresenter * const QtReflMainWindowView::getRunsPresenter() {
    return m_runsPresenter;
}
```
 Wouldn't this imply that my view is slightly more aware of what's outside? I know it is already using the pointers to create the presenter, but it does not need to "keep track" of them. I guess that is the reason why I initially did this.